internal/app/singularity/actions.go

```diff
@@ -310,6 +310,14 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 	engineConfig := singularity.NewConfig()
 
 	configurationFile := buildcfg.SYSCONFDIR + "/singularity/singularity.conf"
+	configFile, err := os.Stat(configurationFile)
```

Comment:
The whole check could be moved around this line : https://github.com/sylabs/singularity/blob/d61d61e2594c50ac3e1e0611c16a087c2f57d09f/internal/pkg/runtime/engines/singularity/prepare.go#L417